steps/24_categorical_support/step.md

```diff
@@ -0,0 +1,143 @@
+- More on how this is tackled in each design below.
+---
+---
+
+
+### Design 1. Allow categorical features ONLY in those estimators which support natively. (no encoding internally)
+
+- `categorical_features` parameter will be present only in estimators that support natively.
+- For estimators that do not support natively, users can make a pipeline with an encoder of their choice and use categorical variables. This is the general ml workflow, like how it's done in sklearn too.
+    
+    #### How tag is going to be used
+    - in datatype checks :
+    ```
+    if categorical support tag is false:
+        if categorical feature is present:
+            error - cat feature is not supported by this estimator
+    ```       
+
+- In this case `categorical feature` will not be a parameter in models that do not support. So (NO,YES,_) is not possible.    
+
+| **Estimator supports categorical natively** | **Cat feature specified by user** | **Cat feature present in data** | **Outcome** | 
+|:-----------------:|:-------------------------:|:--------------:|:---------------:|
+| NO | NO | NO | normal case, proceeds as usual without error. |
+| NO | NO | YES| informative error message to user that categorical feature is not supported by the estimator |
+| NO | YES| NO | X |
+| NO | YES| YES| X |
+| YES| NO | NO | normal case, proceeds as usual without error. |
+| YES| NO | YES| use categorical columns as detected by feature_kind and proceed. |
+| YES| YES| NO | informative error message that specified column was not present in data.|
+| YES| YES| YES| use categorical columns as detected by feature_kind along with the user specified columns(union) and proceed. |
+
+- in case where specified column was not found, we can also choose to ignore and continue with warning that column was not found.
+
+#### estimators that can be added:
+- xgboost, lightgbm and catboost are some models that have native support for categorical. But method of passing categorical features varies quite a lot between them (I've commented on this in umbrella issue - https://github.com/sktime/sktime/issues/6109). Making a general interface might be challenging.
+- So it might be better to interface them in separate classes inheriting from reduction forecaster class. (similar to darts lightgbm, see - https://unit8co.github.io/darts/generated_api/darts.models.forecasting.lgbm.html)
+
+#### PROS:
+- No internal encoding, comparatively easier to implement.
+- No major changes to core logic.
+- Follows general ML workflow i.e if estimator does not support, then user must preprocess their data(including encoding) and then use it.
+#### CONS:
+- Less number of estimators support categorical natively.
+
+
+---
+---
+
+### Design 2. Perform one hot encoding if not natively supported. 
+
+- `categorical_feature`  param will be present in all models.
```

Comment:
Thoughts: if this is available as a configuration instead of as an argument, will that be feasible? If so, will it help?

Reply:
@Abhay-Lejith @fkiraly any thoughts about this comment?